### PR TITLE
Ticket02

### PR DESCRIPTION
**Pull request by LuciaMezquida**
Added a function that receives two parameters: the threshold number of students and an array with the arrival times of the students. If the number of on-time students is less than the threshold number of students,  It must return YES (the class is cancelled), or NO otherwise.
https://www.hackerrank.com/challenges/angry-professor/problem

### CODE REVIEW

**Comment by ferminhg** [comment]
Could you add the link to the problem to understand it better?

File: hackerank/js-practice/angry-professor.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+"use strict";
+
+// Complete the angryProfessor function below.
+function angryProfessor(studentsThreshold, arrivalTime) {
+  const arrivedStudents = arrivalTime.filter((studentTime) => studentTime <= 0);
+  if (arrivedStudents.length < studentsThreshold) {
+    return "YES";
```
Comment (ferminhg) [inline hackerank/js-practice/angry-professor.js]: I recommend extracting `"YES"` and `"NO"` to a constant, and we can reuse late in code.

File: hackerank/js-practice/angry-professor.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+}
+
+function main(studentsThreshold, arrivalTime, response) {
+
+  let result = angryProfessor(studentsThreshold, arrivalTime);
+  if (result === response) {
+    console.log("ok");
+  } else if (result !== response) console.log("not ok");
+}
+main(3, [-1, -2, 1, 3, 4], 'YES');
```
Comment (ferminhg) [inline hackerank/js-practice/angry-professor.js]: Could you add an example where the response is `"NO"`?

File: hackerank/js-practice/angry-professor.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  }
+  return "NO";
+}
+
+function main(studentsThreshold, arrivalTime, response) {
+
+  let result = angryProfessor(studentsThreshold, arrivalTime);
+  if (result === response) {
+    console.log("ok");
+  } else if (result !== response) console.log("not ok");
```
Comment (ferminhg) [inline hackerank/js-practice/angry-professor.js]: You don't need to add the conditional `if (result !== response)`. 

The else explicit that the `result` is not equal to `response`.

**Review by ferminhg — COMMENTED** [review]
Congrats for the naming of variables !!! 👏👏👏👏

All the comments are nitpicking!!! 💯

File: hackerank/js-practice/angry-professor.js (hunk 1 above)
Comment (LuciaMezquida) [reply hackerank/js-practice/angry-professor.js]: I agree 👍🏻

File: hackerank/js-practice/angry-professor.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+function main(studentsThreshold, arrivalTime, response) {
+  let result = angryProfessor(studentsThreshold, arrivalTime);
+  if (result === response) {
+    console.log("ok");
+  } else {
+    console.log("not ok");
+  }
+}
+main(3, [-1, -2, 1, 3, 4], "YES");
+main(4, [-1, -2, 0, -3, 4], "NO");
```
Comment (EleonoraDellamico) [inline hackerank/js-practice/angry-professor.js]: You did great, Lucia! I have read the problem and thank you to the variables' names I could understand better the solution.

File: hackerank/js-practice/angry-professor.js [hunk 5]
```diff
@@ -2,18 +2,23 @@
 
 // Complete the angryProfessor function below.
 function angryProfessor(studentsThreshold, arrivalTime) {
+  const canceledClass = "YES";
```
Comment (ferminhg) [inline hackerank/js-practice/angry-professor.js]: You can define the constant outside the scope of the function and you could reuse it on lines :23 :24

**Review by ferminhg — COMMENTED** [review]
Please, don't resolve comments started by another teammate 🙏

**Pull request merged**